common/lib/core-interfaces/src/serializer.ts

```diff
@@ -28,12 +28,23 @@ export interface IFluidSerializer extends IProvideFluidSerializer {
      *
      * The original `input` object is not mutated.  This method will shallowly clones all objects in the path from
      * the root to any replaced handles.  (If no handles are found, returns the original object.)
+     *
+     * This will be deprecated and replaced in the future with encode.
      */
     replaceHandles(value: any, bind: IFluidHandle): any;
 
+    /**
+     * Given a mostly-plain object that may have handle objects embedded within, will return a fully-plain object
+     * where any embedded IFluidHandles have been replaced with a serializable form.
+     *
+     * The original `input` object is not mutated.  This method will shallowly clones all objects in the path from
+     * the root to any replaced handles.  (If no handles are found, returns the original object.)
```

Comment:
If handles are found, I get a new object, but if not, I get the same object back? Is that expected?

Reply:
Yes and yes. It's the existing behaviour of replaceHandles. It would be easy to change the implementation if you have concerns about this.

Reply:
Just FYI - It's the traditional copy-on-write optimization.  I would recommend keeping it.